### PR TITLE
Aop2-55: Send email to supplier contact when new referral created

### PR DESCRIPTION
**Pull request by waleedusman**

### CODE REVIEW

**Comment by waleedusman** [comment]
The email template may change. In which case I'll have to update it, i.e. `new_referral.md`

File: backend/wsgi.py [hunk 1]
```diff
@@ -6,13 +6,21 @@
...
+from flask import jsonify
+from werkzeug.exceptions import HTTPException
 
 
 port = int(os.getenv('PORT', '5000'))
 application = create_app(os.getenv('DM_ENVIRONMENT') or 'development')
 
 dburl = application.config['SQLALCHEMY_DATABASE_URI']
 
+@application.errorhandler(Exception)
```
Comment (waleedusman) [inline backend/wsgi.py]: I added a general catch-all to api endpoints.

Errors should be a suitable http response code + message, not a wall of html.
This error handler will default unhandled exceptions into a 500 (internal server error).

File: backend/app/emails/referrals.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+from flask import current_app
+from flask_login import current_user
+from app.api.services import suppliers
+from .util import render_email_template, send_or_handle_error
+
+
+def send_new_referral_email(referral_id, supplier_code):
+    template = 'new_referral.md'
+    subject = current_app.config['NEW_REFERRAL_EMAIL_SUBJECT']
```
Comment (bsigney) [inline backend/app/emails/referrals.py]: as discussed (and requested) with Chris (P.O), please add the referral id in the email subject.
see comments in https://ato-orams.atlassian.net/browse/AOP2-55

Comment (bsigney) [inline backend/app/emails/referrals.py]: as discussed (and requested) with Chris (P.O), please add the referral id in the email subject.
see comments in https://ato-orams.atlassian.net/browse/AOP2-55

File: backend/app/api/services/suppliers.py [hunk 3]
```diff
@@ -29,3 +29,11 @@ def find_suppliers(self, prefix):
             .order_by(Supplier.name)
 
         return [supplier.serializable for supplier in suppliers.all()]
+
+    def get_email_address(self, supplier_code):
+        supplier, contact = db.session.query(Supplier, Contact) \
```
Comment (bsigney) [inline backend/app/api/services/suppliers.py]: Is this confirmed that we use the Supplier's first contact to send email to and not the email shown on the service matrix?

**Comment by waleedusman** [comment]
I have updated this PR with the new email design, please review.

**Pull request merged**